File: lib/models/suggestion.dart
class Suggestion {
  final String id;
  final String landmarkName;
  final double latitude;
  final double longitude;
  final String status;
  final String submittedBy;

  Suggestion({
    required this.id,
    required this.landmarkName,
    required this.latitude,
    required this.longitude,
    required this.status,
    required this.submittedBy,
  });

  factory Suggestion.fromJson(Map<String, dynamic> json) {
    return Suggestion(
      id: json['_id'],
      landmarkName: json['landmark_name'],
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      status: json['status'],
      submittedBy: json['submitted_by'] is Map ? json['submitted_by']['full_name'] : json['submitted_by'],
    );
  }
}

File: backend/src/models/Suggestion.js
const mongoose = require('mongoose');

const suggestionSchema = new mongoose.Schema({
  landmark_name: { type: String, required: true },
  latitude: { type: Number, required: true },
  longitude: { type: Number, required: true },
  submitted_by: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  status: { type: String, enum: ['pending', 'approved', 'rejected'], default: 'pending' },
  submitted_at: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Suggestion', suggestionSchema);
